### ISSUE TITLE
Upgrade to v1.12.0 breaks "newrelic_alert_channel" data source implementatiom

### ISSUE THREAD

**Issue by SiddhantAgarwal17** [terraform-providers/terraform-provider-newrelic#366]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

### Affected Resource(s)
Please list the resources as a list, for example:
- newrelic_alert_channel data source

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```
data "newrelic_alert_channel" "email_alert_channels" {
  name  = format("%s_alert_channel", "service-name")
}
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
https://gist.github.com/SiddhantAgarwal17/ae9e7b02884e280e02ab564abb7d11e4
### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
It should have matched with the service-name_alert_channel format as with v1.10.0 and earlier.

### Actual Behavior
It failed to match with the name even though the channel with that name exists in New Relic.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?
No

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234

**Comment by ctrombley** [comment]
@SiddhantAgarwal17 Various problems have been resolved with the `newrelic_alert_channel` resource in the last week.  Please upgrade to the latest version of the provider (v1.12.2 at the time of this comment) and let us know if you are still experiencing problems.

**Comment by SiddhantAgarwal17** [comment]
@ctrombley  I checked again using v1.12.2, but I get the same issue. Here is the output from the `terraform init` command:

`
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "newrelic" (terraform-providers/newrelic) 1.12.2...
`

**Comment by SiddhantAgarwal17** [comment]
I get the following error again

```
module.alert_policies.data.http.service_apps: Refreshing state...
module.alert_policies.data.newrelic_alert_channel.email_alert_channels[0]: Refreshing state...

Error: the name 'service-name_alert_channel' does not match any New Relic alert channel

  on policies/main.tf line 4, in data "newrelic_alert_channel" "email_alert_channels":
   4: data "newrelic_alert_channel" "email_alert_channels" {
```

**Comment by SiddhantAgarwal17** [comment]
@ctrombley  Can this issue be re-opened or do I need to a log an issue again?

**Comment by csMACnz** [comment]
I also had this issue, I had to version lock to the previous version to get it working again.

**Comment by SiddhantAgarwal17** [comment]
Updates please?

**Comment by ctrombley** [comment]
@SiddhantAgarwal17 I am trying to reproduce this locally.

**Comment by piechu** [comment]
Hi,

I had same issue - playing with newrelic-client-go shows that with multiple data, when there are more than 1 page returned by New Relic - data is being overwritten and lost. I've reported it newrelic-client-go project https://github.com/newrelic/newrelic-client-go/issues/117.

Cheers,
Kuba

**Comment by sanderblue** [comment]
@SiddhantAgarwal17 I haven't been able to reproduce this. Any chance you can provide a bit more context, such as more of your config? Even though you're seeing an error that indicates an issue with the data source, I'm starting to wonder if this could be coming from somewhere else under the hood. We have hundreds and hundreds of channels and this seems to be working, which makes me think we might need more information to reproduce.

**Comment by sanderblue** [comment]
Quick follow-up, @piechu's discovery seems to be correct. We'll be working to get this fixed. Thanks for the report.

**Comment by nitesh-babylon** [comment]
I am having issues relating to "newrelic_alert_channel" which may be related to this

I am able to provision a channel for a policy initially but when applying a second time terraform the tool shows a new value has been added for url, where it has been unchanged

This leads to it failing as it "doesn't support update"

```
Terraform will perform the following actions:

  # newrelic_alert_channel.slack will be updated in-place
  ~ resource "newrelic_alert_channel" "slack" {
        id   = "3150950"
        name = "[TF] nitesh-slack-1"
        type = "slack"

      ~ config {
            channel = "#devex-alerts"
            headers = (sensitive value)
            payload = (sensitive value)
          + url     = (sensitive value)
        }
    }```

**Comment by ctrombley** [comment]
@nitesh-babylon The "doesn't support update" issue is being tracked in #372.  We expect drift for channel configurations that contain sensitive values, and the solution for that is to include a `lifecycle` block that eliminates diffs for the `config` block:

```
  lifecycle {
    ignore_changes = ["config"]
  }
```

However, the "doesn't support updates" error is a regression, and we will be pushing up a fix shortly.

**Comment by zlesnr** [comment]
The `doesn't support update` message should be resolved in master.  We'll have a release out this week to include that fix.

**Comment by SiddhantAgarwal17** [comment]
Any update on the original issue in this post? Do we have an expected release date?

**Comment by ctrombley** [comment]
@SiddhantAgarwal17 This issue has been resolved in [v1.13.1](https://github.com/terraform-providers/terraform-provider-newrelic/releases/tag/v1.13.1).  Please let us know if you run into other issues.
